### PR TITLE
[AOSP-pick] [querysync] Fix building dependencies when workspace root is included

### PR DESCRIPTION
**Pull request by LeFrosch**
Cherry pick AOSP commit [43bfac8a806e125bdea33a7270f359fc1454b33b](https://cs.android.com/android-studio/platform/tools/adt/idea/+/43bfac8a806e125bdea33a7270f359fc1454b33b).

NOTE: conflicts with our changes to `aspect/build_dependencies.bzl`, but as far as I can tell ours handles more edge cases.

Change-Id: I2d28bf336e95c410da5f159b6bc37a38b7139f72
Bug: 309706901
Tests: manual and presubmit tests

AOSP: 43bfac8a806e125bdea33a7270f359fc1454b33b

### CODE REVIEW

File: querysync/javatests/com/google/idea/blaze/qsync/AspectUnitTest.java [hunk 1]
```diff
@@ -96,4 +96,15 @@ public void each_jar_belongs_to_a_single_target() throws IOException {
...
   }
+
+  @Test
+  public void workspaceRootIncluded_getTargetSources() throws IOException {
+    ImmutableMap<Label, JavaTargetArtifacts> byTarget =
+        Maps.uniqueIndex(
+            JavaInfoTxt.WORKSPACE_ROOT_INCLUDED.readOnlyProto().getArtifactsList(),
+            t -> Label.of(t.getTarget()));
+    TestData target = TestData.WORKSPACE_ROOT_INCLUDED_QUERY;
+    // no sources should be there for a target within project scope.
```
Comment (tpasternak) [inline querysync/javatests/com/google/idea/blaze/qsync/AspectUnitTest.java:107]: This might looks surprising, but I guess we assume we get it from `bazel query` so no need to get from the aspect.

FTR https://github.com/bazelbuild/intellij/pull/7039/files#diff-5a9bbef8c9fe9d68bc7a4d681f6335be8f023af28f5dbbd292dcdbd6d402c1c3R56
